targets/XPlatCppSdk/source/code/source/playfab/OneDSWinHttpPlugin.cpp

```diff
@@ -0,0 +1,111 @@
+#include <stdafx.h>
+
+#ifndef DISABLE_ONEDS_API
+
+#include <playfab/OneDSWinHttpPlugin.h>
+#include <windows.h>
+#include <winhttp.h>
+
+#pragma warning (disable: 4100) // formal parameters are part of a public interface
+#pragma warning (disable: 4245) // Some DWORD arguments of WinHTTP API can accept -1, according to documentation
+
+namespace PlayFab
+{
+    std::string OneDSWinHttpPlugin::GetUrl(CallRequestContainer& requestContainer) const
+    {
+        return std::string("https://self.events.data.microsoft.com/OneCollector/1.0/");
+    }
+
+    void OneDSWinHttpPlugin::SetPredefinedHeaders(CallRequestContainer& requestContainer, HINTERNET hRequest)
+    {
+        OneDSCallRequestContainer& reqContainer = (OneDSCallRequestContainer&)requestContainer;
+
+        WinHttpAddRequestHeaders(hRequest, L"Accept: */*", -1, 0);
+        WinHttpAddRequestHeaders(hRequest, L"Client-Id: NO_AUTH", -1, 0);
+        WinHttpAddRequestHeaders(hRequest, L"Content-Type: application/bond-compact-binary", -1, 0);
+        WinHttpAddRequestHeaders(hRequest, L"Expect: 100-continue", -1, 0);
+        WinHttpAddRequestHeaders(hRequest, L"SDK-Version: EVT-PlayFab-XPlat-C++-No-3.0.275.1", -1, 0);
+        int64_t now = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch()).count();
+        WinHttpAddRequestHeaders(hRequest, (L"Upload-Time: " + std::to_wstring(now)).c_str(), -1, 0);
+        WinHttpAddRequestHeaders(hRequest, (L"Content-Length: " + std::to_wstring(reqContainer.requestBinaryBody.size())).c_str(), -1, 0);
+        WinHttpAddRequestHeaders(hRequest, L"Connection: Keep-Alive", -1, 0);
```

Comment:
>    WinHttpAddRequestHeaders(hRequest, L"Connection: Keep-Alive", -1, 0); [](start = 4, length = 73)

Can you please point me to the documentation ? We should strive to use the keepalive feature to reuse the http connections

Reply:
That header was literally taken from Aria's SDK and our plugin is a substitute for it.

---
In reply to: [244868108](https://github.com/PlayFab/SDKGenerator/pull/368#discussion_r244868108) [](ancestors = 244868108)

Reply:
I actually found a nice and simple explanation here:
[https://en.wikipedia.org/wiki/HTTP_persistent_connection](https://en.wikipedia.org/wiki/HTTP_persistent_connection)


---
In reply to: [244875698](https://github.com/PlayFab/SDKGenerator/pull/368#discussion_r244875698) [](ancestors = 244875698,244868108)